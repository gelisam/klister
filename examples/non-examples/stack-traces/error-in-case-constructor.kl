#lang "prelude.kl"

(import "lambda-case.kl")

(datatype (Alphabet)
  (a) (b) (c) (d) (e) (f) (g) (h) (i) (j) (k) (l) (m) (n)
  (o) (p) (q) (r) (s) (t) (u) (v) (w) (x) (y) (z) (æ) (ø) (å))

(define fail
  (lambda (thing)
    (case thing
      [(l) (error 'Im-an-error)]
      [(else the-other) the-other])))

(example (fail (l)))
