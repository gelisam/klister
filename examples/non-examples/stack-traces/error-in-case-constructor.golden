In Phase: p0:
    error-in-case-constructor.kl:12.20-12.31: Im-an-error
    stack trace:
    ---- in data case pattern: l
    ---- Halt
